### PR TITLE
Generate versioninfo for some binaries

### PR DESCRIPTION
**Pull request by MDrakos**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-748" title="DX-748" target="_blank">BugDX-748</a>  CLI - PROC: Missing metadata for "state-svc" and "state" processes
  </td></table>
  <br />

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -105,6 +104,7 @@ jobs:
         shell: pwsh
         run: |
           echo "${PSScriptRoot}/.github/deps/${{ runner.os }}/bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+          state run install-deps-ci
```
Comment (mitchell-as) [inline .github/workflows/build.yml:107]: Can we make this the only change in the file? The change on line 261 makes me nervous.

File: activestate.yaml [hunk 2]
```diff
@@ -67,25 +67,44 @@ scripts:
         echo "actionlint was not found on your PATH, installing .."
         go install github.com/rhysd/actionlint/cmd/actionlint@v1.6.23
       fi
+      if { [[ "$GOOS" == "windows" ]] || [[ "$OS" == "Windows_NT" ]]; } && ! type "goversioninfo" &> /dev/null; then
+        echo "actionlint was not found on your PATH, installing .."
```
Comment (mitchell-as) [inline activestate.yaml:71]: This echo is incorrect.

File: activestate.yaml [hunk 3]
```diff
@@ -67,25 +67,44 @@ scripts:
...
     language: bash
     if: ne .OS.Name "Linux"
     description: Install OS specific deps
     value: ""
+  - name: install-deps-ci
+    language: bash
+    if: ne .Shell "cmd"
+    value: |
+      if { [[ "$GOOS" == "windows" ]] || [[ "$OS" == "Windows_NT" ]]; } && ! type "goversioninfo" &> /dev/null; then
+        echo "actionlint was not found on your PATH, installing .."
```
Comment (mitchell-as) [inline activestate.yaml:84]: This echo is also incorrect.

File: activestate.yaml [hunk 4]
```diff
@@ -67,25 +67,44 @@ scripts:
...
+        go run scripts/versioninfo-generator/main.go version.txt cmd/state-svc/versioninfo.json "State Service"
+      fi
   - name: build
     language: bash
     description: Builds the project with the host OS as the target OS.
     value: |
       set -e
       $constants.SET_ENV
 
+      pushd cmd/state > /dev/null
```
Comment (mitchell-as) [inline activestate.yaml:105]: Please add a comment that this is to generate syso from versioninfo.txt (if that's indeed what it does).

File: activestate.yaml [hunk 5]
```diff
@@ -104,6 +123,9 @@ scripts:
       set -e
       $constants.SET_ENV
 
+      pushd cmd/state-svc > /dev/null
```
Comment (mitchell-as) [inline activestate.yaml:126]: Please add a similar comment.

File: internal/assets/contents/versioninfo.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    "FileType": "01",
+    "FileSubType": "00"
+  },
+  "StringFileInfo": {
+    "Comments": "",
+    "CompanyName": "ActiveState Software Inc.",
+    "FileDescription": "",
+    "FileVersion": "",
+    "InternalName": "",
+    "LegalCopyright": "© ActiveState Software, Inc. 2024",
```
Comment (mitchell-as) [inline internal/assets/contents/versioninfo.json:25]: Are we going to have to remember to change this every year? If so, please change the generator to modify this field. Perhaps omit the year here and append it there.

File: scripts/versioninfo-generator/main.go [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+func run() error {
+	if len(os.Args) != 4 {
+		return errs.New("Usage: versioninfo-generator <version.txt> <versioninfo.json>")
+	}
+
+	versionFile := os.Args[1]
+	versionInfoFilePath := os.Args[2]
+	productName := os.Args[3]
+
+	// Read version.txt
```
Comment (mitchell-as) [inline scripts/versioninfo-generator/main.go:98]: I really like the comments in this function!

**Review by mitchell-as — CHANGES_REQUESTED** [review]
This is cool! There are a few minor things to address before merging.

File: .github/workflows/build.yml (hunk 1 above)
Comment (MDrakos) [reply .github/workflows/build.yml:108]: Bah, that's my damn YAML formatter.

**Review by mitchell-as — APPROVED** [review]
Looks great!

**Pull request merged**